gamedata/alldefs_post.lua

```diff
@@ -1054,6 +1054,9 @@ function WeaponDef_Post(name, wDef)
 		-- wDef.targetborder = 1.0
 
 		if wDef.weapontype == "Cannon" then
+			-- do not cast shadows on projectiles
```

Comment:
I just mean skipping this if a model is provided, because then the projectile is no longer glowing plasma.
```suggestion
		if wDef.weapontype == "Cannon" and not wDef.model then
			-- do not cast shadows on glowing plasma
```